### PR TITLE
Test unlikely cases of ECC modular reduction

### LINKED ISSUE #7109: Test unlikely cases of ECC modular reduction
Optimized reductions for arithmetic on elliptic curves tends to have unlikely cases, such as an intermediate value being between P and 2^n where P is very close to 2^n.

For example, after https://github.com/Mbed-TLS/mbedtls/pull/6971, the `LAST` step of `mbedtls_ecp_mod_p192_raw` executes `*p = c`, and at this point it's very unlikely that `c != 0` and we don't have a unit test that causes `c` to be nonzero. If we forgot this assignment, or wrote `c` at the wrong offset, or mistakenly removed it, that wouldn't cause a unit test to fail.

The goal of this task is to investigate where we want to have such test cases, figure out how to construct suitable input values, and implement the tests. This should probably be broken into further subtasks after some initial analysis.

Also part of the investigation is the definition of done. When do we have enough coverage? I'd say at a minimum when removing any single statement would make the code wrong, but this may not be a good measure.

### PR DESCRIPTION
**Pull request by gabor-mezei-arm**
## Description

Resolve #7109

## Gatekeeper checklist

- [x] **changelog** Not required: no user visible changes
- [X] **backport** Not required: not present in 2.28
- [x] **tests** provided

### CODE REVIEW

**Review by minosgalanakis — COMMENTED** [review]
Looks good overall, one minor comment for a p521 test case.

File: scripts/mbedtls_dev/ecp.py [hunk 1]
```diff
@@ -327,7 +379,15 @@ class EcpP521R1Raw(bignum_common.ModOperationCommon,
...
 
-        # Corner case: maximum canonical P521 multiplication result
+        # Modulus - 1
+        ("01ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+         "fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe"),
+
+        # Modulus + 1
+        ("020000000000000000000000000000000000000000000000000000000000000000"
+         "000000000000000000000000000000000000000000000000000000000000000000"),
+
```
[the remark this answers is not in the document]
Comment (gabor-mezei-arm) [reply scripts/mbedtls_dev/ecp.py:389]: Not needed because `P521 = 2^521 - 1`

**Review by minosgalanakis — APPROVED** [review]
